src/query/script/tests/it/testdata/script-error.txt

```diff
@@ -9,39 +9,51 @@ error:
 
 
 ---------- Input ----------
-LET x := 1;
 LET x RESULTSET := SELECT 1;
+LET y := x;
 ---------- Output ----------
 error: 
-  --> SQL:2:5
+  --> SQL:2:10
   |
-1 | LET x := 1;
-2 | LET x RESULTSET := SELECT 1;
-  |     ^ `x` is already defined as a different kind of variable
+1 | LET x RESULTSET := SELECT 1;
+2 | LET y := x;
```

Comment:
x is `ResultSet`, so y could be `ResultSet` too ?

ResultSet could be cloned.

Reply:
y can be resultset, as long as y is declared as `LET y RESULTSET := x`

Reply:
Ok, so the default type is Scalar.